code/modules/codex/entries/_codex_entry.dm

```diff
@@ -63,10 +63,10 @@
 	if(!name)
 		if(length(associated_strings))
 			name = associated_strings[1]
-		else
+		else if(store_codex_entry)
 			CRASH("Attempted to instantiate unnamed codex entry with no associated strings!")
 
-	if(store_codex_entry)
+	if(name && store_codex_entry)
```

Comment:
I feel like this can be simplified or changed somehow, but it should work fine as-is